src/CADL.Extension/Emitter.Csharp/src/options.ts

```diff
@@ -64,7 +76,9 @@ const defaultOptions = {
     "generate-protocol-methods": true,
     "generate-convenience-methods": true,
     "package-name": undefined,
-    debug: undefined
+    debug: undefined,
+    "treat-empty-string-as-null-for-models": undefined,
```

Comment:
why one is undefined, and one is empty array?

Reply:
Originally, both default values could be undefined as if users don't set them, we don't want to store them in the `Configuration.json` and undefined is the right value to check if users have set them. 

But for `treat-empty-string-as-null-for-extra-types`, we need to union it with the default set of types, using `[]` is easier without additional check. 

Besides, whether we should store the full list of types depends on whether `treat-empty-string-as-null-for-models` is undefined since if the model list is empty then the types apply to no models and there is also no use to only store the default list of types. 

Therefore, the default value for the types config doesn't need to be undefined.